table/server/master/src/main/java/alluxio/master/table/UnpartitionedTableScheme.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.table;
+
+import alluxio.grpc.table.FieldSchema;
+import alluxio.grpc.table.Layout;
+
+import com.google.common.base.Preconditions;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Unpartitoned table scheme.
+ */
+public class UnpartitionedTableScheme implements PartitionScheme {
+  private Partition mPartition;
+
+  /**
+   * Constructor for UnpartitionedTableScheme.
+   *
+   * @param partitions a list of partitions
+   */
+  public UnpartitionedTableScheme(List<Partition> partitions) {
+    Preconditions.checkArgument(partitions.size() == 1);
+    mPartition = partitions.get(0);
+  }
+
+  @Override
+  public List<Partition> getPartitions() {
+    return Arrays.asList(mPartition);
```

Comment:
can we save this as a `final` member variable?

Reply:
done